tests/contrib/hooks/test_gcp_api_base_hook.py

```diff
@@ -44,6 +44,66 @@
     default_creds_available = False
 
 
+class NoForbiddenAfterCount:
+    """Holds counter state for invoking a method several times in a row."""
+
+    def __init__(self, count, **kwargs):
+        self.counter = 0
+        self.count = count
+        self.kwargs = kwargs
+
+    def __call__(self):
+        """
+        Raise an Forbidden until after count threshold has been crossed.
+        Then return True.
+        """
+        if self.counter < self.count:
+            self.counter += 1
+            raise Forbidden(**self.kwargs)
+        return True
+
+
+@hook.GoogleCloudBaseHook.quota_retry(wait=tenacity.wait_none())
+def _retryable_test_with_temporare_quota_retry(thing):
+    return thing()
+
+
+class QuotaRetryTestCase(unittest.TestCase):  # ptlint: disable=invalid-name
+    def test_do_nothing_on_non_error(self):
+        result = _retryable_test_with_temporare_quota_retry(lambda: 42)
+        self.assertTrue(result, 42)
+
+    def test_retry_on_exception(self):
+        message = "POST https://translation.googleapis.com/language/translate/v2: User Rate Limit Exceeded"
+        errors = [
+            {
+                'message': 'User Rate Limit Exceeded',
+                'domain': 'usageLimits',
+                'reason': 'userRateLimitExceeded',
+            }
+        ]
+        try:
+            _retryable_test_with_temporare_quota_retry(NoForbiddenAfterCount(
+                count=5,
+                message=message,
+                errors=errors
+            ))
+            self.assertTrue(True)  # pylint: disable=redundant-unittest-assert
```

Comment:
I'd still want to assert the count in NoForbiddenAfterCount object. here and not assert Exception. It is not really useful (hiding exception stack trace + not asserting anything).